contrib/tools/meta.md/eqform.md

```diff
@@ -565,5 +579,145 @@ fmod EQFORM-SET is
   op (_|_) : EqFormNeSet     EqFormSet     -> EqFormNeSet    [ctor ditto] .
   op (_|_) : NormFormNeSet   NormFormSet   -> NormFormNeSet  [ctor ditto] .
   op (_|_) : FormNeSet       FormSet       -> FormNeSet      [ctor ditto] .
+
+  var S : Substitution . var NeSS : NeSubstitutionSet . var SS : SubstitutionSet .
+  var F : Form         . var FNeS : FormNeSet         . var FS : FormSet .
+
+  op _<<_ : FormSet SubstitutionSet -> [FormSet] .
+  ------------------------------------------------
+  eq mtFormSet  << SS = mtFormSet .
+  eq (F | FNeS) << SS = (F << SS) | (FNeS << SS) .
+
+  eq FS << .SubstitutionSet = mtFormSet .
+  eq FS << (S | NeSS)       = (FS << S) | (FS << NeSS) .
+endfm
+```
+
+```maude
+fmod EQFORM-OPERATIONS is
+  pr EQFORM .
+  pr TERM-EXTRA . --- defines vars() : Term -> QidSet
+
+  vars F F1 F2 : Form . vars PEC1 : PosEqConj .
+  vars M : Module . vars T T' : Term .
+
+  op  wellFormed : Module Form -> [Bool] .
+  op $wellFormed : Module Form -> [Bool] .
+  ----------------------------------------
+ ceq  wellFormed(M,F) = $wellFormed(M,F) if wellFormed(M) .
+
+  eq $wellFormed(M,tt)       = true .
+  eq $wellFormed(M,ff)       = true .
+  eq $wellFormed(M,~ F)      = $wellFormed(M,F) .
+ ceq $wellFormed(M,F1 /\ F2) = $wellFormed(M,F1) and-then $wellFormed(M,F2) if F1 =/= tt /\ F2 =/= tt .
+ ceq $wellFormed(M,F1 \/ F2) = $wellFormed(M,F1) and-then $wellFormed(M,F2) if F1 =/= ff /\ F2 =/= ff .
```

Comment:
I believe (but that belief may be wrong) that the extra subsorts I added to the formula structure should make these kinds of conditional equations unnecessary (which show up a couple of times in this file, not just above). Did you try unconditional equations with subsorts and have termination/correctness issues?

Reply:
@skeirik2 I've switched over to using the unconditional equations over the subsorts, and the rewrite counts went down. Nice!

Reply:
LGTM.